framework/kms-keyring.md

```diff
@@ -309,7 +312,84 @@ If the response is successfully verified, OnDecrypt MUST do the following with t
+independently decrypt the resulting encrypted message.
+
+For example, if a user configures a KMS keyring with CMKs A, B, and C
+and uses that keyring to encrypt a message,
+then that user expects three things to be true:
+
+1. CMK A can decrypt the encrypted message.
+1. CMK B can decrypt the encrypted message.
+1. CMK C can decrypt the encrypted message.
+
+If any of these are not true of the resulting encrypted message,
+the keyring has failed to honor the user's intent.
+
+In order to accomplish this,
+the KMS keyring MUST contribute three encrypted data keys to the encryption materials:
+one from CMK A, one from CMK B, and one from CMK C.
+
+### OnDecrypt Goal
+
+When a user configures a KMS keyring for use on decrypt,
+they are stating their intent for which CMKs
+the keyring will *attempt* to use to decrypt encrypted data keys.
+
+For example, if a user configures a KMS keyring with CMK C
+and uses it to decrypt an encrypted message
+that contains encrypted data keys for CMKs A, B, and C,
+then the keyring will attempt to decrypt using CMK C.
+
+However, if the keyring attempts to decrypt using CMK C and cannot,
+this failure still honors the configured intent and MUST NOT halt decryption.
+The configured intent is that they keyring MUST *attempt* with these CMKs,
+not that they MUST *succeed*.
+
+### Why OnEncrypt and OnDecrypt are different
+
+On encrypt, the user describes their intent
+for the requirements to decrypt the resulting message.
+Because of this,
+the keyring MUST create encryption materials that satisfies those requirements.
+
+On decrypt, the user provides resources that *attempt* to do that decryption.
+
+This is an asymmetric relationship with very different implications on failure.
+If the keyring encounters a problem on encrypt,
+it cannot fully honor the decryption requirements and so MUST halt message encryption.
+
+However, on decrypt the keyring is not creating anything.
+It is instead attempting to satisfy the requirements that were set on encryption.
+If the keyring cannot satisfy those requirements it MUST NOT halt message decryption.
+
+No keyring can know if it is the last keyring to attempt decryption.
+If all keyrings are exhausted and none of them were able to decrypt an encrypted data key
+then the cryptographic materials manager that managed those keyrings
+will halt message decryption.
+(See [Default Cryptographic Materials Manager](./default-cmm.md))
+
+### Requirements
+
+These goals can be reduced to the following two requirements:
+
+1. On encrypt, if any configured CMK cannot be used,
+    that is an error and encryption MUST halt.
+1. On decrypt, the keyring MUST never halt decryption because of a failure to decrypt.
+
 ## Security Considerations
 
 [TODO: What security properties are guaranteed by this keyring? Also, note how the security properties
 can vary drastically depending on key policies]
+
+## Contributing Issues
```

Comment:
I like this idea, but it will be easy to forget to follow up on unless we're all on board. Maybe we could update the PR template to specifically call out to include such updates in the spec. (Doesn't have to be done as part of this PR, just a thought)

Reply:
Agreed. I created #77 to track this.